forui/lib/src/widgets/accordion.dart

```diff
@@ -0,0 +1,399 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'dart:math' as math;
+
+import 'package:forui/forui.dart';
+import 'package:forui/src/foundation/tappable.dart';
+import 'package:forui/src/foundation/util.dart';
+import 'package:meta/meta.dart';
+
+/// A controller that stores the expanded state of an [FAccordion].
+class FAccordionController extends ChangeNotifier {
+  late final AnimationController _animation;
+  late final Animation<double> _expand;
+
+  /// Creates a [FAccordionController].
+  FAccordionController({
+    required TickerProvider vsync,
+    duration = const Duration(milliseconds: 500),
+    bool initiallyExpanded = true,
+  }) {
+    _animation = AnimationController(
+      duration: duration,
+      value: initiallyExpanded ? 1.0 : 0.0,
+      vsync: vsync,
+    );
+    _expand = Tween<double>(
+      begin: 0,
+      end: 100,
+    ).animate(
+      CurvedAnimation(
+        curve: Curves.ease,
+        parent: _animation,
+      ),
+    )..addListener(notifyListeners);
+  }
+
+  /// Convenience method for toggling the current [expanded] status.
+  ///
+  /// This method should typically not be called while the widget tree is being rebuilt.
+  Future<void> toggle() async => expanded ? close() : expand();
+
+  /// Expands the accordion.
+  ///
+  /// This method should typically not be called while the widget tree is being rebuilt.
+  Future<void> expand() async {
+    await _animation.forward();
+    notifyListeners();
+  }
+
+  /// closes the accordion.
+  ///
+  /// This method should typically not be called while the widget tree is being rebuilt.
+  Future<void> close() async {
+    await _animation.reverse();
+    notifyListeners();
+  }
+
+  /// True if the accordion is expanded. False if it is closed.
+  bool get expanded => _animation.value == 1.0;
+
+  /// The animation value.
+  double get value => _expand.value;
+
+  @override
+  void dispose() {
+    _animation.dispose();
+    super.dispose();
+  }
+}
+
+/// A vertically stacked set of interactive headings that each reveal a section of content.
+///
+/// See:
+/// * https://forui.dev/docs/accordion for working examples.
+class FAccordion extends StatefulWidget {
+  /// The accordion's style. Defaults to the appropriate style in [FThemeData.accordionStyle].
+  final FAccordionStyle? style;
+
+  /// The title.
+  final String title;
+
+  /// The accordion's controller.
+  final FAccordionController? controller;
+
+  /// Whether the accordion is initially expanded. Defaults to true.
+  final bool initiallyExpanded;
```

Comment:
Why is initiallyExpanded declared on both the widget and controller?

Reply:
I added this to allow the user to set the `initiallyExpanded` field of the default controller,  only if they did not add in a their own controller. 

After your question, I was wondering if adding some new comments will help to make that clearer.
  /// Whether the accordion is initially expanded. Defaults to true.
  ///
  /// This flag will be ignored if a [controller] is provided.

Reply:
I'm kinda on the fence about this. On one hand, I agree it's much more convenient to configure it via a bool param. On the other, specifying it in both the controller and widget seems confusing.

Perhaps we can make the field nullable, add an assertion in the constructor to ensure that only either param is set?

Reply:
Also, Maybe i should change the controller field to be required? since the widget already provides a default value?